### PR TITLE
Improvements to some counsel-find-file actions

### PR DESCRIPTION
**Pull request by ericdanan**
See commit messages.

### CODE REVIEW

File: counsel.el [hunk 1]
```diff
@@ -1767,7 +1767,10 @@ choose between `yes-or-no-p' and `y-or-n-p'; otherwise default to
             :caller 'counsel-find-file-move))
 
 (defun counsel-find-file-mkdir-action (_x)
-  (make-directory (expand-file-name ivy-text ivy--directory)))
+  (make-directory (expand-file-name ivy-text ivy--directory))
+  (unless (eq ivy-exit 'done)
+    (with-selected-window (active-minibuffer-window)
+      (ivy--cd (expand-file-name ivy-text ivy--directory)))))
```
Comment (basil-conto) [inline counsel.el]: I think it's safer to check that `active-minibuffer-window` does not return `nil`:

```el
(defun counsel-find-file-mkdir-action (_x)
  (let ((dir (expand-file-name ivy-text ivy--directory))
        (win (and (not (eq ivy-exit 'done))
                  (active-minibuffer-window))))
    (make-directory dir)
    (when win (with-selected-window win (ivy--cd dir)))))
```

Please also add a docstring. Thanks.

File: counsel.el [hunk 2]
```diff
@@ -1754,7 +1754,9 @@ choose between `yes-or-no-p' and `y-or-n-p'; otherwise default to
             (counsel--yes-or-no-p "Delete %s? " x))
     (dired-delete-file x dired-recursive-deletes delete-by-moving-to-trash)
     (dired-clean-up-after-deletion x)
-    (ivy--reset-state ivy-last)))
+    (unless (eq ivy-exit 'done)
+      (with-selected-window (active-minibuffer-window)
```
Comment (basil-conto) [inline counsel.el]: Ditto re: `active-minibuffer-window` returning `nil`.

File: counsel.el [hunk 3]
```diff
@@ -1740,11 +1740,12 @@ choose between `yes-or-no-p' and `y-or-n-p'; otherwise default to
...
   "Copy file X."
-  (let ((ivy-inhibit-action
-         (lambda (new-name)
-           (require 'dired-aux)
-           (dired-copy-file x new-name 1))))
-    (counsel-find-file)))
+  (counsel--find-file-1 "Copy file to: "
+                        ivy--directory
+                        (lambda (new-name)
+                          (require 'dired-aux)
```
Comment (basil-conto) [inline counsel.el]: The `require` should be moved to the toplevel of the function, I think.

File: counsel.el [hunk 4]
```diff
@@ -1760,13 +1761,12 @@ choose between `yes-or-no-p' and `y-or-n-p'; otherwise default to
...
-            :matcher #'counsel--find-file-matcher
-            :action (lambda (new-name)
-                      (require 'dired-aux)
-                      (dired-rename-file x new-name 1))
-            :keymap counsel-find-file-map
-            :caller 'counsel-find-file-move))
+  (counsel--find-file-1 "Rename file to: "
+                        ivy--directory
+                        (lambda (new-name)
+                          (require 'dired-aux)
```
Comment (basil-conto) [inline counsel.el]: Ditto re: hoisting `require` out of the action function.

**Comment by ericdanan** [comment]
Thanks for the review, I applied your suggestions.

**Comment by abo-abo** [comment]
Thanks, all.

**Pull request closed without merging**